### PR TITLE
test: make test TestTopSQLCPUProfile stable by mock high cpu usage (#25702)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #25702 to release-5.1
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/25771
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/25771:release-5.1-9e8bb505e11d
```

---

Signed-off-by: crazycs <crazycs520@gmail.com>

### What problem does this PR solve?

```sql
[2021-06-23T05:48:54.266Z] ----------------------------------------------------------------------
[2021-06-23T05:48:54.266Z] FAIL: tidb_test.go:1196: tidbTestTopSQLSuite.TestTopSQLCPUProfile
[2021-06-23T05:48:54.266Z] 
[2021-06-23T05:48:54.266Z] tidb_test.go:1385:
[2021-06-23T05:48:54.266Z]     checkFn(ca.sql, ca.planRegexp)
[2021-06-23T05:48:54.266Z] tidb_test.go:1365:
[2021-06-23T05:48:54.266Z]     // since 1 sql may has many plan, check `len(stats) > 0` instead of `len(stats) == 1`.
[2021-06-23T05:48:54.266Z]     c.Assert(len(stats) > 0, IsTrue, commentf)
[2021-06-23T05:48:54.266Z] ... obtained bool = false
[2021-06-23T05:48:54.266Z] ... sql: select * from t where a in (1,2,3,4
```

### What is changed and how it works?

Some SQL is run so fast that pprof profile unable to sample those SQL, this PR use failpoint to inject some cpu load for each SQL to make sure each SQL can be sampled by Top SQL.

### Related changes

- N/A

### Check List 

Tests 

- No code

Side effects

- N/A

### Release note 

- No release note.

### CODE REVIEW

File: server/tidb_test.go [hunk 1]
```diff
@@ -1252,9 +1292,16 @@ func (ts *tidbTestTopSQLSuite) TestTopSQLCPUProfile(c *C) {
 		{prepare: "insert into t1 (b) values (?);", args: []interface{}{1}, planRegexp: ""},
 		{prepare: "replace into t1 (b) values (?);", args: []interface{}{1}, planRegexp: ""},
 		{prepare: "update t1 set b=a where b is null limit ?;", args: []interface{}{1}, planRegexp: ".*Limit.*TableReader.*"},
+<<<<<<< HEAD
```
Comment (zimulala) [inline server/tidb_test.go:1295]: @crazycs520 
Please fix this conflict.

**Review by zimulala — APPROVED** [review]
LGTM

**Review by AilinKid — APPROVED** [review]
LGTM

**Pull request merged**